extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildComplexMetricSerde.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.datasketches.hll;
+
+import java.nio.ByteBuffer;
+
+import org.apache.druid.data.input.InputRow;
+import org.apache.druid.segment.column.ColumnBuilder;
+import org.apache.druid.segment.data.GenericIndexed;
+import org.apache.druid.segment.data.ObjectStrategy;
+import org.apache.druid.segment.serde.ComplexColumnPartSupplier;
+import org.apache.druid.segment.serde.ComplexMetricExtractor;
+import org.apache.druid.segment.serde.ComplexMetricSerde;
+
+public class HllSketchBuildComplexMetricSerde extends ComplexMetricSerde
+{
+
+  @Override
+  public String getTypeName()
+  {
+    return HllSketchModule.TYPE_NAME; // must be common type name
+  }
+
+  @Override
+  public ObjectStrategy getObjectStrategy()
+  {
+    return HllSketchObjectStrategy.STRATEGY;
+  }
+
+  @Override
+  public ComplexMetricExtractor getExtractor()
+  {
+    return new ComplexMetricExtractor()
```

Comment:
There are many extractors in the project identical to this, please extract a static utility.

Reply:
only two, and they are different

Reply:
How they are different? This one will be the third. I think three repetitions justifiy adding an utility.

Reply:
`SketchBuildComplexMetricSerde`, `ArrayOfDoublesSketchBuildComplexMetricSerde`.  `HllSketchBuildComplexMetricSerde` is the third

Reply:
Oh, you are looking across different modules. Yes, this build extractor can be factored out, which would save a few lines of trivial code.